File: backend/src/models/POSCategory.js
const mongoose = require('mongoose');

const posCategorySchema = new mongoose.Schema({
  key: {
    type: String,
    required: true,
    unique: true,
    uppercase: true,
    trim: true,
    enum: ['BUS', 'CANTEEN', 'VENDING_MACHINE']
  },
  name: {
    type: String,
    required: true,
    trim: true
  },
  icon: {
    type: String,
    required: true,
    trim: true
  },
  description: {
    type: String,
    trim: true
  },
  isActive: {
    type: Boolean,
    default: true
  },
  displayOrder: {
    type: Number,
    default: 0
  }
}, {
  timestamps: true
});

posCategorySchema.index({ key: 1 }, { unique: true });
posCategorySchema.index({ displayOrder: 1 });

module.exports = mongoose.model('POSCategory', posCategorySchema);

File: mobile-app/nfc_app/lib/features/pos/domain/pos_category.dart
class POSCategory {
  const POSCategory({
    required this.id,
    required this.key,
    required this.name,
    required this.icon,
    this.description,
    this.isActive = true,
    this.displayOrder = 0,
  });

  final String id;
  final String key;
  final String name;
  final String icon;
  final String? description;
  final bool isActive;
  final int displayOrder;

  factory POSCategory.fromJson(Map<String, dynamic> json) {
    return POSCategory(
      id: json['_id']?.toString() ?? json['id']?.toString() ?? '',
      key: json['key']?.toString() ?? '',
      name: json['name']?.toString() ?? '',
      icon: json['icon']?.toString() ?? '',
      description: json['description']?.toString(),
      isActive: json['isActive'] as bool? ?? true,
      displayOrder: (json['displayOrder'] as num?)?.toInt() ?? 0,
    );
  }
}
